sklearn/neighbors/classification.py

```diff
@@ -155,18 +156,37 @@ def predict(self, X):
         weights = _get_weights(neigh_dist, self.weights)
 
         y_pred = np.empty((n_samples, n_outputs), dtype=classes_[0].dtype)
+
+        if self._issparse:
+            y_pred_sparse = []
+
         for k, classes_k in enumerate(classes_):
```

Comment:
I think you want two separate code-paths for this routine. Putting the condition inside the loop makes the code pretty messy.